### ISSUE TITLE
Compiler bug on conditional assignment with multiple value and an optional return type

### ISSUE THREAD

**Issue by dedesite** [vlang/v#14441]
**V doctor:**
```
OS: linux, Ubuntu 20.04.4 LTS
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
CC version: cc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
V full version: V 0.2.4 0e5c1ce.65d9c8f

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg bug.v`
```v
fn consume() ?int {
	return 32
}

fn main() {
	operation, val := if true {
		'add', consume()
	} else {
		'sub', consume()
	}
}

```

**What did you expect to see?**

A proper compilation error like "error: consume() returns an option, so it should have either an `or {}` block, or `?` at the end

**What did you see instead?**
```
==================
/tmp/v_1000/bug.13318921018810689572.tmp.c:1140: error: field 'arg1' has incomplete type
==================
```

**Comment by KamkoAmoh** [comment]
yeah, I agree, but the problem (not sure) comes on how it fails to optional on execution, by extracting `consume` outside `if,else` it will throw you proper error, example:

```vlang
fn consume() ?int {
	return 32
}

fn main() {
		con:= consume() ?

	operation, val := if true {
		'add', con
	} else {
		'sub', con
	}
}
```

**Comment by dedesite** [comment]
Thanks @KamkoAmoh I know that my code was not good, but I filled a bug because I got a compiler bug and not a proper error ;).

**Comment by kolts1** [comment]
Pull request #15428

**Comment by StunxFS** [comment]
@medvednikov @spytheman this is fixed by #15428 , close it please

**Comment by spytheman** [comment]
Done.
